server/datastore/redis/redistest/redistest.go

```diff
@@ -73,11 +73,30 @@ func SetupRedis(tb testing.TB, cluster, redir, readReplica bool) fleet.RedisPool
 	require.Nil(tb, err)
 
 	tb.Cleanup(func() {
-		err := redis.EachNode(pool, false, func(conn redigo.Conn) error {
-			_, err := conn.Do("FLUSHDB")
-			return err
-		})
-		require.NoError(tb, err)
+		if cleanupKeyPrefix != "" {
+			keys, err := redis.ScanKeys(pool, cleanupKeyPrefix+"*", 1000)
+			require.NoError(tb, err)
+			for _, k := range keys {
+				func() {
+					conn := pool.Get()
+					defer conn.Close()
+					if _, err := conn.Do("DEL", k); err != nil {
+						require.NoError(tb, err)
+					}
+				}()
+			}
+		} else {
+			// NOTE: it's definitely best to avoid this - can create random failures for other
```

Comment:
You could print a `WARNING` message that will be printed when running the Go tests.

Reply:
Alternatively have SetupRedis use the `t.Name()` as prefix (and not have the `cleanupKeyPrefix ` argument).

Reply:
Though at some point, I kind of like that idea that all redis tests would use such a well-defined prefix, but I'm not sure are doable it is at the moment (all redis uses need to support somehow injecting a test prefix and use it in all places). Certainly not for this PR, but we could investigate this for the future - one option I would've loved is to have the redis client support a "key prefix" option and use it automatically on all commands, so you'd just set it on the redis setup and automatically all calls with that pool would be sandboxed to that prefix. In practice it's not that simple, as not all redis commands use a key, or have the keys in the same positions...

Reply:
> I was thinking, how about we fail the test if there's no prefix specified? It will clearly bite us later if there's a test that does this, so might as well catch it ASAP.

+1!